wrappers/services.go

```diff
@@ -969,17 +992,31 @@ func StopServices(apps []*snap.AppInfo, flags *StopServicesFlags, reason snap.Se
 			}
 		}
 
+		// Is the service slot activated, then lets warn the user this doesn't have any
+		// real effect if a disable was requested
+		if flags.Disable && serviceIsSlotActivated(app) {
+			logger.Noticef("Disabling %s may not have the intended effect as the service is currently always activated by a slot", app.Name)
+		}
+
+		// Get services including any activation mechanisms. When stopping and disabling
+		// services we do it on both the primary service, and it's activation mechanisms. The
+		// StartServices logic does actually not enable/start any service which are activated,
+		// but rather only the activation services themselves, so one might argue if it
+		// is really necessary to disable the primary service.
+		svcs := serviceUnitsIncludingActivations(app)
+
 		var err error
 		timings.Run(tm, "stop-service", fmt.Sprintf("stop service %q", app.ServiceName()), func(nested timings.Measurer) {
-			err = stopService(sysd, app, inter)
+			err = stopService(sysd, inter, app.DaemonScope, svcs)
 			if err == nil && flags.Disable {
-				disableServices = append(disableServices, app.ServiceName())
+				disableServices = append(disableServices, svcs...)
```

Comment:
Do you mean we should just disable them instead of adding them to the list?

Reply:
Not in place disabling, rather that `disableServices := []string{}` can be removed and on line 1011, `flags.Disable` can be used to test if `svcs` should be disabled, instead of `len(disableServices) > 0`.